gr-blocks/lib/selector_impl.cc

```diff
@@ -149,8 +151,20 @@ int selector_impl::general_work(int noutput_items,
     const uint8_t** in = (const uint8_t**)&input_items[0];
     uint8_t** out = (uint8_t**)&output_items[0];
 
+
     gr::thread::scoped_lock l(d_mutex);
     if (d_enabled) {
+        auto nread = nitems_read(d_input_index);
+        auto nwritten = nitems_written(d_output_index);
+
+        std::vector<tag_t> tags;
+        get_tags_in_window(tags, 0, 0, noutput_items);
+
+        for (auto tag : tags) {
+            tag.offset -= (nread - nwritten);
```

Comment:
This should be correct, however, I can not test right now.

Me suggestion from #3572 to simply use `tag.offset = nitems_written(d_output_index)` is wrong. The tag offset must be modified to `offset at input − number of dropped samples = offset at input − (number items read − number items written)` (as used in this code). Or put another way to `number items written + relative offset at input = number items written + (offset at input − number items read)`.